views/login.html

```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Login</title>
+  </head>
+  <body>
+    <form action="/login" method="POST">
```

Comment:
In order to authenticate via jwt Bearer header token you should here use ajax call in order to 'capture' the token received from the login response.

Here's a bot generated example. Put it inside <script> tag and modify accordingly.

```
// Get the form element
const form = document.querySelector('#yourIdForm');

// Add event listener to the form's submit event
form.addEventListener('submit', handleSubmit);

// Handle form submission
function handleSubmit(event) {
  event.preventDefault(); // Prevent default form submission behavior

  const formData = new FormData(form); // Get the form data

  fetch('/login, {
    method: 'POST',
    body: formData // Set the form data as the request body
  })
  .then(response => response.json()) // Parse the response as JSON
  .then(data => {
    // Handle the response data
    console.log(data);
   // save to local storage (read more here: https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage)
   localStorage.setItem('user-data', data);
  })
  .catch(error => {
    // Handle any errors
    console.error('Error:', error);
  });
}
```

Reply:
Now, after successful login you will have the token in the localStorage, and use it when you need to create/update a link.

In order to do so you will need to change the code in 'index.html' by submitting the form in ajax there as well:
```
const token = localStorage.getItem('user-data');
// remember localstorage always serializes data to string, so my code from the login then handler is wrong because we pass object. It needs to be stringified before saving there

// Handle form submission
function handleLinkFormSubmit(event) {
  event.preventDefault(); // Prevent default form submission behavior

  const formData = new FormData(form); // Get the form data

  fetch('/api/shorturl, {
    method: 'POST',
    headers: {
        'Authorization': `Bearer ${token}`
    }
    body: formData // Set the form data as the request body
  })
  .then()
   ...
}
```